### PR TITLE
Program to print ancestors of given number of BST

### PR DESCRIPTION
**Pull request by vrandachag**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes!

### Description

Program to print Ancestors of given number of BST in C++.

### Checklist

- [Y] I've read the contribution guidelines.
- [Y] I've checked the issue list before deciding what to submit.
- [Y] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

(#1565 issue is solved here)

### CODE REVIEW

File: C-Plus-Plus/ds/Ancestors of given number in BST.cpp [hunk 1]
```diff
@@ -0,0 +1,129 @@
+#include<iostream>
```
Comment (HarshCasper) [inline C-Plus-Plus/ds/Ancestors of given number in BST.cpp]: Add a beginner pitch enclosed in a multiline comment detailing about the Code/Algorithm

Comment (HarshCasper) [inline C-Plus-Plus/ds/Ancestors of given number in BST.cpp]: Name the file properly. Like in your case, it should not be `Ancestors of given number in BST.cpp` but `ancestors_of_given_number_BST.cpp`. Make it look more meaningful and understanding.

File: C-Plus-Plus/ds/Ancestors of given number in BST.cpp [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+using namespace std;
+
+struct BST
+{
+	int a,count = 0, level = 0; 
+	BST *left = NULL;
+	BST *right = NULL;
+};
+
+BST *start = NULL; //points to the root node
```
Comment (HarshCasper) [inline C-Plus-Plus/ds/Ancestors of given number in BST.cpp]: Put Comments above the Line of Code not beside it

File: C-Plus-Plus/ds/Ancestors of given number in BST.cpp [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+int l1,l2;
+
+BST *arr_to_BST(int arr[20], int beg, int end) //function to convert array into Binary Search Tree with shortest height
```
Comment (HarshCasper) [inline C-Plus-Plus/ds/Ancestors of given number in BST.cpp]: Put Comments above the Line of Code not beside it

File: C-Plus-Plus/ds/Ancestors of given number in BST.cpp [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+		int mid = (beg+end)/2;
+		ptr->left = arr_to_BST(arr,beg,mid-1);
+		ptr->right = arr_to_BST(arr,mid+1,end);	
+		return ptr;
+	}
+	else
+		return NULL;
+}
+
+void Display_BST(BST *ptr) //function to print Binary Search Tree
```
Comment (HarshCasper) [inline C-Plus-Plus/ds/Ancestors of given number in BST.cpp]: Put Comments above the Line of Code not beside it

File: C-Plus-Plus/ds/Ancestors of given number in BST.cpp [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+		cout<<"\tright:"<<ptr->right->a;
+	
+	if(ptr->left != NULL)
+		Display_BST(ptr->left);
+	
+	if(ptr->right != NULL)
+		Display_BST(ptr->right);
+}
+
+void level_assign(BST *ptr, int value=0) //function to assign level values to corresponding nodes
```
Comment (HarshCasper) [inline C-Plus-Plus/ds/Ancestors of given number in BST.cpp]: Put Comments above the Line of Code not beside it

File: C-Plus-Plus/ds/Ancestors of given number in BST.cpp [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+		find_number(ptr->left,n);
+	else
+		find_number(ptr->right,n);
+}
+
+void Display(BST *ptr, int n1, int n2) //function to print the value of youngest member for given numbers
+{
+	if(n1 < ptr->a && n2 > ptr->a)
+	{
+		cout<<"\nyoungest:"<<ptr->a;
```
Comment (HarshCasper) [inline C-Plus-Plus/ds/Ancestors of given number in BST.cpp]: Use meaningful Console Statements

**Review by HarshCasper — CHANGES_REQUESTED** [review]

**Comment by vrandachag** [comment]
I've done the mentioned changes. Please have a look.

File: C-Plus-Plus/ds/ancestors_of_given_number_BST.cpp [hunk 7]
```diff
@@ -0,0 +1,140 @@
+/*
```
Comment (HarshCasper) [inline C-Plus-Plus/ds/ancestors_of_given_number_BST.cpp:1]: Can you add a brief summary about what Youngest and Oldest Members means? We usually add a description keeping in mind that the reader knows nothing about the same.

File: C-Plus-Plus/ds/ancestors_of_given_number_BST.cpp [hunk 8]
```diff
@@ -0,0 +1,140 @@
...
+		int mid = (beg+end)/2;
+		ptr->left = arr_to_BST(arr,beg,mid-1);
+		ptr->right = arr_to_BST(arr,mid+1,end);	
+		return ptr;
+	}
+	else
+		return NULL;
+}
+
+void Display_BST(BST *ptr) //function to print Binary Search Tree
```
Comment (HarshCasper) [inline C-Plus-Plus/ds/ancestors_of_given_number_BST.cpp]: Put Comments above the Line of Code. Not beside it.

File: C-Plus-Plus/ds/ancestors_of_given_number_BST.cpp [hunk 9]
```diff
@@ -0,0 +1,140 @@
...
+		else
+		{
+			if(n1 < start->a && n2 > start->a)
+				;
+			else
+				Display(start, n1, n2);
+		}
+	}
+	return 0;
+}
```
Comment (HarshCasper) [inline C-Plus-Plus/ds/ancestors_of_given_number_BST.cpp:143]: Add sample I/O at the end of the Code.

**Review by HarshCasper — CHANGES_REQUESTED** [review]
Rest looks good! Once done, ping me on Slack and I will merge it.

**Comment by HarshCasper** [comment]
LGTM!

**Pull request merged**